### PR TITLE
ROX-21641: compliance result stat cluster

### PR DESCRIPTION
**Pull request by dashrews78**
## Description

This PR calculates and groups the check result stats by cluster.  So if a cluster had 14 different scan configurations run against it, this would aggregate the counts of all 14 of those results into a single count.  Below is an instance of this with 3 completely different scans executed against the same cluster.  The stats are aggregated.

```
{
   "scanStats" : [
      {
         "checkStats" : [
            {
               "count" : 406,
               "status" : "FAIL"
            },
            {
               "count" : 0,
               "status" : "INFO"
            },
            {
               "count" : 288,
               "status" : "PASS"
            },
            {
               "count" : 0,
               "status" : "ERROR"
            },
            {
               "count" : 35,
               "status" : "MANUAL"
            },
            {
               "count" : 0,
               "status" : "INCONSISTENT"
            },
            {
               "count" : 0,
               "status" : "NOT_APPLICABLE"
            }
         }
         ],
         "cluster" : {
            "clusterId" : "454ef133-efa6-4b83-9762-c82cb9723360",
            "clusterName" : "remote"
         }
      }
   ]
}
```

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Added unit tests.

Also executed against a cluster with compliance operator installed, where multiple profiles were run against a single cluster.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by bradr5 — APPROVED** [review]

File: proto/api/v2/compliance_results_service.proto [hunk 1]
```diff
@@ -105,6 +111,11 @@ message ListComplianceScanResultsResponse {
     repeated ComplianceScanResult scan_results = 1;
 }
 
+// ListComplianceCheckScanStatsResponse provides stats based on the check names
```
Comment (theencee) [inline proto/api/v2/compliance_results_service.proto:114]: provides overall states per cluster? Where does check names come into play here?

ComplianceCheckStatusCount -> maps cluster to list of ComplianceCheckStatusCount

ComplianceCheckStatusCount -> count per status (PASS, FAIL). I don't see a check name

File: central/complianceoperator/v2/checkresults/datastore/datastore.go [hunk 2]
```diff
@@ -24,7 +24,10 @@ type DataStore interface {
 	SearchComplianceCheckResults(ctx context.Context, query *v1.Query) ([]*storage.ComplianceOperatorCheckResultV2, error)
 
 	// ComplianceCheckResultStats retrieves the scan results stats specified by query
-	ComplianceCheckResultStats(ctx context.Context, query *v1.Query) ([]*ResourceCountByResultByCluster, error)
+	ComplianceCheckResultStats(ctx context.Context, query *v1.Query) ([]*ResourceResultCountByClusterScan, error)
+
+	// ComplianceClusterStats retrieves the scan results stats specified by query
```
Comment (theencee) [inline central/complianceoperator/v2/checkresults/datastore/datastore.go:29]: The comment is a copy of the previous one and doesn't make sense

File: central/complianceoperator/v2/checkresults/datastore/datastore_impl.go [hunk 3]
```diff
@@ -108,8 +97,51 @@ func (d *datastoreImpl) ComplianceCheckResultStats(ctx context.Context, query *v
...
+			{
+				Field: search.Cluster.String(),
+			},
+		}
+	}
+
+	countQuery := d.withCountByResultSelectQuery(cloned, search.ClusterID)
+	countResults, err := pgSearch.RunSelectRequestForSchema[ResultStatusCountByCluster](ctx, d.db, schema.ComplianceOperatorCheckResultV2Schema, countQuery)
 	if err != nil {
+		log.Info(err)
```
Comment (theencee) [inline central/complianceoperator/v2/checkresults/datastore/datastore_impl.go:144]: log.Error? (or wrap the err and return?)

**Review by theencee — APPROVED** [review]
Minor comments about comments

**Comment by bradr5** [comment]
Are we able to get operator status and type included in the cluster data as well?

File: proto/api/v2/compliance_results_service.proto (hunk 1 above)
Comment (dashrews78) [reply proto/api/v2/compliance_results_service.proto:114]: Check name doesn't come into play here at all.  I copied the comment from another WIP PR.

**Review by bradr5 — APPROVED** [review]

**Pull request merged**